src/platforms/pipedrive/schemas.ts

```diff
@@ -0,0 +1,234 @@
+            start: z.number().optional(),
+          })
+          .optional(),
+      })
+      .optional(),
+  });
+export type SearchOutput<T> = {
+  data?: {
+    items?: {
+      result_score: number;
+      item?: T;
+    }[];
+  };
+  additional_data?: {
+    properties?: { more_items_in_collection?: boolean; start?: number };
+  };
+};
+export const searchResponseSchema = (itemSchema: z.ZodSchema) =>
+  z.object({
+    data: z
+      .object({
+        items: z
+          .array(
+            z.object({
+              result_score: z.number(),
+              item: itemSchema,
+            }),
+          )
+          .optional(),
+      })
+      .optional(),
+    additional_data: z
+      .object({
+        properties: z
+          .object({
+            more_items_in_collection: z.boolean().optional(),
+            start: z.number().optional(),
+          })
+          .optional(),
+      })
+      .optional(),
+  });
+
+export const pipedriveModuleSchema = z.enum(PIPEDRIVE_MODULES);
+export type PipedriveModule = (typeof PIPEDRIVE_MODULES)[number];
+
+export const upsertResponseSchema = z.object({
+  id: z.string(),
+});
+export type UpsertResponse = {
+  id: string;
+};
+
+// -
+// Subtypes
+// -
+export type PipedriveEmailAddressType = (typeof EMAIL_TYPES)[number];
+export const pipedriveEmailAddressSchema = z.object({
+  primary: z.boolean(),
+  label: z.string().transform((l) => l as PipedriveEmailAddressType),
+  value: z.string(),
+});
+
+export type PipedrivePhoneNumberType = (typeof PHONE_TYPES)[number];
+export const pipedrivePhoneNumberSchema = z.object({
+  primary: z.boolean(),
+  label: z.string().transform((l) => l as PipedrivePhoneNumberType),
+  value: z.string(),
+});
+
+export const pipedriveAssociationId = z.object({
+  value: z.number(),
+  name: z.string(),
+});
+
+// -
+// Users
+// -
+export const pipedriveUserSchema = z.object({
+  id: z.number(),
+  name: z.string(),
+  email: z.string(),
+  created: custom.date(),
+  modified: custom.date().optional(),
+  active_flag: z.boolean(),
+});
+export type PipedriveUser = z.infer<typeof pipedriveUserSchema>;
+
+// -
+// Person
+// -
+const pipedrivePersonSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  first_name: z.string(),
+  last_name: z.string(),
+  email: z.array(pipedriveEmailAddressSchema),
+  phone: z.array(pipedrivePhoneNumberSchema),
+  add_time: custom.date(),
+  update_time: custom.date(),
+  active_flag: z.boolean(),
+  org_id: pipedriveAssociationId.optional(),
+  owner_id: pipedriveAssociationId.optional(),
+});
+export type PipedrivePerson = z.infer<typeof pipedrivePersonSchema>;
+
+export const pipedrivePersonUpsertSchema = z
+  .object({
+    name: z.string(),
+    first_name: z.string(),
+    last_name: z.string(),
+    email: z.array(pipedriveEmailAddressSchema),
+    phone: z.array(pipedrivePhoneNumberSchema),
+    org_id: z.number().optional(),
+    owner_id: z.number().optional(),
+    $native: z.record(z.any()),
```

Comment:
this shouldn't have the $native property right? Since this is the zod schema for the client?

Reply:
We do include `$native` in any upsert schemas so that the typing makes sense when you're using it in the unified api, and there is a `formatUpsertInputWithNative` that removes this field before passing it to the request. (the other clients are doing it like this right now)

In hindsight maybe I should have made it the responsibility of the unified code to spread the $native data, but then again it would be kind of weird to make the typing work for that